packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -11,57 +11,53 @@ import { __ } from '@wordpress/i18n';
 import { useSelect, useDispatch } from '@wordpress/data';
 import { useState } from '@wordpress/element';
 
-const EntitiesSavedStatesCheckbox = ( {
-	id,
-	name,
-	changes: { rawRecord },
-	checked,
-	setCheckedById,
-} ) => (
-	<CheckboxControl
-		label={ `${ name }: "${ rawRecord.name ||
-			rawRecord.slug ||
-			rawRecord.title ||
-			__( 'Untitled' ) }"` }
-		checked={ checked }
-		onChange={ ( nextChecked ) => setCheckedById( id, nextChecked ) }
-	/>
-);
-
 export default function EntitiesSavedStates( {
 	isOpen,
 	onRequestClose,
-	ignoredForSave = new EquivalentKeyMap(),
+	ignoredForSave = [],
 } ) {
-	const entityRecordChangesByRecord = useSelect( ( select ) =>
-		select( 'core' ).getEntityRecordChangesByRecord()
+	const dirtyEntityRecords = useSelect(
+		( select ) => select( 'core' ).__experimentalGetDirtyEntityRecords(),
+		[]
 	);
 	const { saveEditedEntityRecord } = useDispatch( 'core' );
 
-	const [ checkedById, _setCheckedById ] = useState(
-		() => new EquivalentKeyMap()
-	);
-	const setCheckedById = ( id, checked ) =>
-		_setCheckedById( ( prevCheckedById ) => {
-			const nextCheckedById = new EquivalentKeyMap( prevCheckedById );
-			if ( checked ) {
-				nextCheckedById.set( id, true );
-			} else {
-				nextCheckedById.delete( id );
-			}
-			return nextCheckedById;
-		} );
+	const [ unsavedEntityRecords, _setUnsavedEntityRecords ] = useState( [] );
+	const setUnsavedEntityRecords = ( { kind, name, key }, checked ) => {
+		if ( checked ) {
+			_setUnsavedEntityRecords(
+				unsavedEntityRecords.filter(
+					( elt ) =>
+						elt.kind !== kind ||
+						elt.name !== name ||
+						elt.key !== key
+				)
+			);
+		} else {
+			_setUnsavedEntityRecords( [
+				...unsavedEntityRecords,
+				{ kind, name, key },
+			] );
+		}
+	};
 	const saveCheckedEntities = () => {
-		checkedById.forEach( ( _checked, id ) => {
-			if ( ! ignoredForSave.has( id ) ) {
-				saveEditedEntityRecord(
-					...id.filter(
-						( s, i ) => i !== id.length - 1 || s !== 'undefined'
-					)
+		const entitiesToSave = dirtyEntityRecords.filter(
+			( { key, kind, name } ) => {
+				return ! some(
+					ignoredForSave.concat( unsavedEntityRecords ),
+					( elt ) =>
+						elt.kind === kind &&
+						elt.name === name &&
+						elt.key === key
 				);
 			}
+		);
+
+		entitiesToSave.forEach( ( { key, kind, name } ) => {
```

Comment:
```suggestion
		entitiesToSave.forEach( ( { kind, name, key } ) => {
```

It'd be nicer to have all these in this order for consistency.